.github/workflows/build_arrow-incubator.yml

```
@@ -26,4 +26,15 @@ jobs:
         cd $BASEDIR
         git clone https://github.com/arrow-kt/arrow.git
     - name: Build with Gradle
-      run: $BASEDIR/arrow/scripts/project-build.sh $ARROW_LIB
+      run: |
+        $BASEDIR/arrow/scripts/project-build.sh $ARROW_LIB
+    - name: Run benchmarks
+      run: |
+        ./gradlew :arrow-benchmarks-free:jmhForArrow
+        ./gradlew :arrow-benchmarks-free:jmhForCats
+    - name: Run benchmark comparison
+      env:
+        CI_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        export PULL_REQUEST_NUMBER=$(echo $GITHUB_REF | cut -d/ -f3)
+        ./gradlew :arrow-benchmarks-free:compareBenchmarksCI
```

```suggestion
        echo "ref: $GITHUB_REF"
        echo "sha: $GITHUB_SHA"
        export PULL_REQUEST_NUMBER=$(echo $GITHUB_REF | cut -d/ -f3)
        echo "pr-number: $PULL_REQUEST_NUMBER"
        ./gradlew :arrow-benchmarks-free:compareBenchmarksCI
```

I'd need these data to discover the reason of 403 error when creating the comment.